src/potions.ts

```diff
@@ -0,0 +1,211 @@
+
+    let quantityToUse = 0;
+    let embezzlersRemaining = Math.max(embezzlers - haveEffect(this.effect()), 0);
+    let keepGoing = true;
+
+    // Use however many will land entirely on embezzler turns.
+    const embezzlerQuantity = Math.floor(embezzlersRemaining / duration);
+    if (this.net(embezzlersRemaining, doubleDuration) > 0 && embezzlerQuantity > 0) {
+      acquire(
+        embezzlerQuantity,
+        this.potion,
+        this.gross(embezzlersRemaining, doubleDuration),
+        false
+      );
+      quantityToUse = Math.max(embezzlerQuantity, itemAmount(this.potion));
+      print(
+        `Determined that ${quantityToUse} ${this.potion.plural} are profitable on embezzlers.`,
+        "blue"
+      );
+      embezzlersRemaining -= quantityToUse * duration;
+    }
+    if (quantityToUse < embezzlerQuantity || (doubleDuration && quantityToUse > 0)) {
+      keepGoing = false;
+    }
+
+    // Now, is there one with both embezzlers and non-embezzlers?
+    if (keepGoing && this.net(embezzlersRemaining, doubleDuration) > 0 && embezzlersRemaining > 0) {
+      acquire(1, this.potion, this.gross(embezzlersRemaining, doubleDuration), false);
+      const additional = Math.max(1, itemAmount(this.potion) - quantityToUse);
+      print(
+        `Determined that ${additional} ${this.potion.plural} are profitable on partial embezzlers.`,
+        "blue"
+      );
+      quantityToUse += additional;
+      embezzlersRemaining = Math.max(embezzlersRemaining - additional * duration, 0);
+    }
+    if (embezzlersRemaining > 0 || (doubleDuration && quantityToUse > 0)) keepGoing = false;
+
+    // How many should we use with non-embezzlers?
+    if (keepGoing && this.net(0, doubleDuration) > 0) {
+      const adventureCap = myAdventures() * 1.04 + 50;
+      const tourists = adventureCap - haveEffect(this.effect()) - quantityToUse * duration;
+      if (tourists > 0) {
+        const touristQuantity = Math.ceil(tourists / duration);
+        acquire(touristQuantity, this.potion, this.gross(0, doubleDuration), false);
+        const additional = Math.max(touristQuantity, itemAmount(this.potion) - quantityToUse);
+        print(
+          `Determined that ${additional} ${this.potion.plural} are profitable on tourists.`,
+          "blue"
+        );
+        quantityToUse += additional;
+      }
+    }
+
+    if (quantityToUse > 0) {
+      if (doubleDuration) quantityToUse = 1;
+      use(quantityToUse, this.potion);
+    }
+  }
+}
+
+export function potionSetup(): void {
+  // TODO: Count PYEC.
+  // TODO: Count free fights (25 meat each for most).
+  const embezzlers = embezzlerCount();
+  const potions = Item.all().filter(
+    (item) => item.tradeable && !banned.includes(item) && itemType(item) === "potion"
+  );
+  const meatPotions = potions
+    .map((item) => new Potion(item))
+    .filter((potion) => potion.bonusMeat() > 0);
+
+  if (have($item`Eight Days a Week Pill Keeper`) && !get("_freePillKeeperUsed")) {
```

Comment:
Given that we're (presumably) calling this before doing all of our embezzler fights, we want to make sure to only do this if it's more valuable than another embezzler. First, check if we have treasury access (canAdv or questStep), and if we do, we'll want to see how much meat an additional embezzler is worth.

Reply:
If we decide this is too hard to calculate I think it's reasonable to assume that you want to get the embezzler if you can. "I think" is pulling a lot of weight here, though.

Reply:
I put in 10,000 as a guess at the value of an additional embezzler. I should add the canAdv check too.